### PR TITLE
adds active flag to terms defaulting to true

### LINKED ISSUE #149: `active` isn't surfaced in #all, causing unnecessary #find calls
https://github.com/samvera/questioning_authority/blob/master/lib/qa/authorities/local/file_based_authority.rb#L17

If we surface the `active` flag, this would allow the caller to handle selecting the terms much more efficiently. Possibly defaulting to `true` if active doesn't exist in the yml file. 

```
res[:active] = true if res[:active].nil?
{ id: res[:id], label: res[:term], active: res[:active] }.with_indifferent_access
```

This would make the additional `#active?` in Hyrax unnecessary and speed up handling large local authorities. https://github.com/samvera/hyrax/blob/master/app/services/hyrax/qa_select_service.rb#L30

### PR DESCRIPTION
**Pull request by revgum**
fixes #149

### CODE REVIEW

File: lib/qa/authorities/local/file_based_authority.rb [hunk 1]
```diff
@@ -14,7 +14,8 @@ def search(q)
 
     def all
       terms.map do |res|
-        { id: res[:id], label: res[:term] }.with_indifferent_access
+        res[:active] = true if res[:active].nil?
+        { id: res[:id], label: res[:term], active: res[:active] }.with_indifferent_access
```
Comment (jcoyne) [inline lib/qa/authorities/local/file_based_authority.rb]: Can we get rid of line 17 and do `active: res.fetch(:active, true)`?

Comment (revgum) [inline lib/qa/authorities/local/file_based_authority.rb]: @jcoyne oh yea.. of course I had to space that one out. #noob

**Comment by mjgiarlo** [comment]
@revgum I reconfigured Coveralls for this repo such that it only fails when coverage drops 0.3% or more, btw. Should pick that up on the next build.

**Comment by mjgiarlo** [comment]
@jcoyne re-review?

**Pull request merged**